### DISCUSSION TITLE
v7.2.6 Release of PowerShell

### DISCUSSION THREAD

**Discussion by adityapatwardhan** [PowerShell/PowerShell#17877] [General]
### Engine Updates and Fixes

- Fix `ForEach-Object -Parallel` when passing in script block variable (#16564)

### General Cmdlet Updates and Fixes

- Make `Out-String` and `Out-File` keep string input unchanged (#17455)
- Update regex used to remove ANSI escape sequences to be more specific to decoration and hyperlinks (#16811)
- Fix legacy `ErrorView` types to use `$host.PrivateData` colors (#17705)
- Fix `Export-PSSession` to not throw error when a rooted path is specified for `-OutputModule` (#17671)

### Tests

- Disable RPM SBOM test. (#17532)

### Build and Packaging Improvements

<details>

<summary>
<p>Bump .NET SDK to 6.0.8 (Internal 22065)</p>
<p>We thank the following contributors!</p>
<p>@tamasvajk</p>
</summary>

<ul>
<li>Update Wix manifest</li>
<li>Add AppX capabilities in MSIX manifest so that PS7 can call the AppX APIs (#17416)</li>
<li>Use <code>Quality</code> only with <code>Channel</code> in <code>dotnet-install</code> (#17847)</li>
<li>Fix <code>build.psm1</code> to not specify both version and quality for <code>dotnet-install</code> (#17589) (Thanks @tamasvajk!)</li>
<li>Install .NET 3.1 as it is required by the vPack task</li>
</ul>

</details>

[7.2.6]: https://github.com/PowerShell/PowerShell/compare/v7.2.5...v7.2.6

### SHA256 Hashes of the release artifacts

- hashes.sha256
  - A17F7AABB18C96DB502D3A443D62B6576972CBA48F965C6FF6342C7552B154AA
- powershell_7.2.6-1.deb_amd64.deb
  - A0D810FE381B77E4BFB99CC67F713F6D483545E94BDEB4150524C085CF20E2DA
- powershell-7.2.6-1.cm.x86_64.rpm
  - 9FA82FD880452E903D45121E036AB114AB22DE3682C57C7E31C89CF6A44853E7
- powershell-7.2.6-1.rh.x86_64.rpm
  - CDA2593C9B360DBFFA5AC42FD836721F9B4CEE0743CD035613CC303C04311149
- powershell-7.2.6-linux-alpine-x64.tar.gz
  - 04281F83816F22074E7CDCB79169DF36F416D35D137251DA554C2DFD5362974B
- powershell-7.2.6-linux-arm32.tar.gz
  - CFCB64387659335C37225CF8E55FC79A810C5AC3685242C310D99C9C56B48FED
- powershell-7.2.6-linux-arm64.tar.gz
  - EAB633392CC4CF989294720CD856977DE194248651595F2E2B73556404497B94
- powershell-7.2.6-linux-x64.tar.gz
  - 9927F6CF4BCBB686074938F8FE662796C10D378A2F6B61E2A9C73E590A1A37D2
- powershell-7.2.6-linux-x64-fxdependent.tar.gz
  - 584C65042E2EBF059EA5C52C98454D75B0B8E70CE74F56775B3203DA0417C16A
- powershell-7.2.6-osx-arm64.pkg
  - 05952E8CA2A4F2633D43EDDAC3A64EE3F1F1EDF90ECBF7EF8B638E4BC72075F0
- powershell-7.2.6-osx-arm64.tar.gz
  - 5E4EC9D67F4DCE90ACFD92645DC359E08A6F07369EF6BA3D93C8DE25C175343E
- powershell-7.2.6-osx-x64.pkg
  - 07CA6DDFBB6A9D98671C2F166CEE93DF2844F2A2C4132F59E71AC9869577564D
- powershell-7.2.6-osx-x64.tar.gz
  - E86AAEAA9C18304A1275B73C2234107D62973EF7934C110E10ECACEBFD11C80E
- PowerShell-7.2.6-win-arm32.zip
  - F0F0414CF74C20D4C41D3EEF31FF9CCD3468B36F4C57BADD5CA44F5854F802D7
- PowerShell-7.2.6-win-arm64.zip
  - 33C09ECC109FDBD446D3EA5509BE5AA831DAFBD47FCC5057C7C7CAB0E503D2B8
- PowerShell-7.2.6-win-fxdependent.zip
  - D9F1F1544DD2C496F5CF0A1B1DC7385ADA69DF45C0064ACE70B063DE3EA68127
- PowerShell-7.2.6-win-fxdependentWinDesktop.zip
  - 29563A9B4B7F34B07A242A42B79FD1F6B7F0581DA4CCC39CE3C67A94F4077636
- PowerShell-7.2.6-win-x64.msi
  - 77566D5B831523C6740C392B64237C52089B574F342A9F51D84ADED24E412CB8
- PowerShell-7.2.6-win-x64.zip
  - 4FB17E905B3422DF8EE748A1A42826996C0741E3C6AFFA41371F19B95404DB38
- PowerShell-7.2.6-win-x86.msi
  - CC6C3B8EECA872C27EF4B42C3CDEB49652112F6CDE72E16D7006F073E543936D
- PowerShell-7.2.6-win-x86.zip
  - 04E0691F7F56388B6396B931CD23AEF6E86504B0AA147660317F342538914551
- powershell-lts_7.2.6-1.deb_amd64.deb
  - FF202D7A1773806DF4D150F860F91028CD318FF5557B99EF43E7B07002D784DF
- powershell-lts-7.2.6-1.cm.x86_64.rpm
  - D56EF4DEAE79D0EA42EB095324DE9897E4C7A904FE13A5D66D8BE6D9AD76D5C5
- powershell-lts-7.2.6-1.rh.x86_64.rpm
  - E8F258E95B4A34C402EF5421EA0C2F1136AC1D100D97E9176E3A82BCC68F7489
- powershell-lts-7.2.6-osx-arm64.pkg
  - 024B6CAACCF9E4C63C836C300496290D96C319CF187E480C8DF10A38DC1E1BB9
- powershell-lts-7.2.6-osx-x64.pkg
  - AF8CC415483AE2881D852B9E046E71BECB6353A6881DADEF35FE253CAACF69B6

<hr /><em>This discussion was created from the release <a href='https://github.com/PowerShell/PowerShell/releases/tag/v7.2.6'>v7.2.6 Release of PowerShell</a>.</em>

**Comment by uchris13** [comment]
Can we update powershell from powershell itself? the below message appeared but it doesnt say how to upgrade 

A new PowerShell stable release is available: v7.2.6
   Upgrade now, or check out the release page at:
     https://aka.ms/PowerShell-Release?tag=v7.2.6

**Reply by johngagefaulkner** [reply]
I would love a `npm install -g npm@latest` for PowerShell.

```powershell
Get-PSLatestVersion -Install -Silent -Force
```

However, there are existing ways to install/upgrade PS7 from PS7. Here's a couple:

```powershell
# The "aka.ms" shortened URL expands to: https://raw.githubusercontent.com/PowerShell/PowerShell/master/tools/install-powershell.ps1

# Method 1: Using Microsoft's "Install-PowerShell" script
## All the parameters and examples are outlined in the top of the script itself.
iex "& { $(irm 'https://aka.ms/install-powershell.ps1') } -UseMSI -Quiet -AddExplorerContextMenu"

# Method 2: Using winget package manager
winget install --id Microsoft.Powershell --source winget

<# 
    - [Other additional methods]
        - [Microsoft Docs | Scripting PowerShell Installation] https://docs.microsoft.com/en-us/powershell/scripting/install/installing-powershell-on-windows?view=powershell-7.2
#>
```

**Reply by iSazonov** [reply]
There is RFC for this https://github.com/PowerShell/PowerShell-RFC/pull/324

**Comment by tata1013** [comment]
thanks

**Comment by Rameezistan** [comment]
chala ja BHSDK behen k loode
